docs/sources/architecture/configuration.md

```diff
@@ -64,17 +71,34 @@ Configuration descriptions must be placed as XML file(s) (with the ending `.xml`
   <tr><td>parameter.required</td><td>Specifies whether the value is required (optional).</td></tr>
   <tr><td>parameter.readOnly</td><td>Specifies whether the value is read-only (optional).</td></tr>
   <tr><td>parameter.multiple</td><td>Specifies whether multiple selections of options are allowed (optional).</td></tr>
+(optional).</td></tr>
+  <tr><td>parameter.advanced</td><td>Specifies that this is an advanced parameter. Advanced parameters may be hidden by a UI (optional).</td></tr>
+(optional).</td></tr>
   <tr><td>context</td><td>The context of the configuration parameter (optional).</td></tr>
   <tr><td>required</td><td>The flag indicating if the configuration parameter has to be set or not (deprecated, optional, default: false).</td></tr>
+  <tr><td>group</td><td>Sets a group name for this parameter (optional).</td></tr>
   <tr><td>default</td><td>The default value of the configuration parameter (optional).</td></tr>
   <tr><td>label</td><td>A human readable label for the configuration parameter (optional).</td></tr>
   <tr><td>description</td><td>A human readable description for the configuration parameter (optional).</td></tr>
   <tr><td>option</td><td>The element definition of a static selection list (optional).</td></tr>
   <tr><td>option.value</td><td>The value of the selection list element.</td></tr>
+  <tr><td>multipleLimit</td><td>If multiple is true, sets the maximum number of options that can be selected (optional).</td></tr>
+  <tr><td>limitToOptions</td><td>If true (default) will only allow the user to select items in the options list. If false, will allow the user to enter other text (optional).</td></tr>
   <tr><td>criteria</td><td>The filter criteria for values of a dynamic selection list (optional).</td></tr>  
   <tr><td>criteria.name</td><td>The name of the context related filter.</td></tr>  
 </table>
 
+Groups allow parameters to be grouped together into logical blocks so that the user can find the parameters they are looking for. A parameter can be placed into a group so that the UI knows how to display the information.
+<table>
+  <tr><td><b>Property</b></td><td><b>Description</b></td></tr>
+  <tr><td>group.name</td><td>The group name - this is used to link the parameters into the group, along with the groupName option in the parameter (mandatory).</td></tr>
```

Comment:
group.name isn't it. It's an attribute called name, inside the group. I think this is consistent with the conventions used elsewhere in the document?